src/Action.ts

```diff
@@ -4,44 +4,127 @@ import * as fs from 'node:fs';
 
 import YAML from 'yaml';
 
+import { type Branding, DEFAULT_BRAND_COLOR, DEFAULT_BRAND_ICON } from './constants.js';
 import LogTask from './logtask/index.js';
 
-export interface InputType {
+/**
+ * Represents an input for the action.
+ */
+export interface Input {
```

Comment:
I don't understand why you declare interfaces instead of types.
I don't see any class implementing this interface.

This comment is valid for many other cases.

Reply:
It is being used in the Action class:
```javascript
  /** Input definitions */
  inputs: { [key: string]: Input };
```

Can you show me how you would write that as a type instead of an interface? Perhaps I am using Interfaces wrong.
I have been using interrfaces instead of types when I want to describe an object's records.

Reply:
In typescript we use to define types to declare "DTO" shape.
I propose to do it in a dedicated PR (This one is huge enough IMO)

Reply:
@neilime is this what you are talking about? https://dev.to/tamj0rd2/dto-a-typescript-utility-type-4o3m